lib/LWP/UserAgent.pm

```diff
@@ -801,13 +806,22 @@ sub cookie_jar {
     if (@_) {
 	my $jar = shift;
 	if (ref($jar) eq "HASH") {
-	    require HTTP::Cookies;
-	    $jar = HTTP::Cookies->new(%$jar);
-	}
-        elsif (ref($jar) eq "ARRAY") {
-            require HTTP::CookieJar::LWP;
-            $jar = HTTP::CookieJar::LWP->new(@$jar);
+        my $class = $self->{cookie_jar_class};
+        try {
+            require Module::Load;
+            Module::Load::load($class);
+	        $jar = $class->new(%$jar);
         }
+        catch {
+            my $error = $_;
+            if ($error =~ /Can't locate/) {
+                die "cookie_jar_class '$class' not found\n";
+            }
```

Comment:
Instead of having a `try` block that can fail in several different ways, and handling those failures conditionally within a single `catch` block, I would break the single try and catch into several, with each `catch` failing in a suitable way.  Additionally, this would mean the try and catch that simply contains `catch { die "$error\n" }` could be removed as the existence of the try and catch behaves the same as its absence.

Reply:
I used the same pattern we have for loading protocol classes in https://github.com/libwww-perl/libwww-perl/blob/master/lib/LWP/Protocol.pm#L69.

Reply:
Having discussed this with Olaf, we've agreed that reusing a construct we already have elsewhere is fine. Thank you for the suggestion though.